astropy/table/table.py

```diff
@@ -1933,10 +2031,15 @@ def replace_column(self, name, col):
         if self[name].info.indices:
             raise ValueError('cannot replace a table index column')
```

Comment:
@taldcroft - Do you think that this check is still needed ? Removing it would allow to fix #8077, and it seems to work if I remove it but I have not checked in detail that the output file is correct.

And btw thanks for these PRs which looks a great improvement for masked tables !

Reply:
I think one issue here is if a column is part of an index with another column, then you can corrupt that index.  Indices in Table was done by a GSoC student and I'll confess to not understanding the details all that well, so my first reaction is always to leave things alone.

Maybe when replacing an index column then just drop the indices that use it?  Users should not be surprised that indices go away if they replace the column.  Thoughts @saimn   @astrofrog ?

Reply:
Ah indeed the multi-column case needs a special treatment. I think it's fine to drop the index in this case, this feature is probably not much used anyway (it is not possible to use a mlti-column index with `.loc`).